### DISCUSSION TITLE
Problema: Vue no actualiza servidor

### DISCUSSION THREAD

**Discussion by stariluz** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#560] [Q&A] [answered]
Ya me di a la tarea de buscar entre todas las preguntas y no encontré ninguna que tratara este problema. 

Contexto: Estoy usando WSL en el cual pude instalar node, npm y vue sin problemas, cree mi proyecto y todo bien.
Ahora, cuando ejecuto
`npm run serve` (que por cierto, a mi parecer tarda demasiado en ejecutar), parece que todo va bien, si tengo errores en el código me los indica. El problema viene al editar los componentes o el app.vue, donde al parecer en el servidor no se reflejan los cambios que realizo. Dejo un seguimiento de capturas de como recreo eso mismo.
![image](https://user-images.githubusercontent.com/53382424/160760558-a0192298-be23-420c-aabf-28307cbdefe3.png)
![image](https://user-images.githubusercontent.com/53382424/160760619-41a93320-7c6f-43ff-a05c-8a5c11efc8a1.png)
![image](https://user-images.githubusercontent.com/53382424/160760771-3ffe96fe-66ba-4584-95d3-aa73a0a131f5.png)

Eso es lo que sale al correr el servidor, con el código que tenía en ese momento, al hacer algún cambio, este no se actualiza en la pagina:
![image](https://user-images.githubusercontent.com/53382424/160760864-e223c31a-a7cf-41ab-8e0b-c97a90dd2cbf.png)
![image](https://user-images.githubusercontent.com/53382424/160760932-783e045c-0eb4-4803-9f6a-d960bf058a3e.png)

Ya investigue un rato y no encontré algo que me sirva (o que entienda jajaja) , seguiré buscando pero si alguno presentó el error agradecería su ayuda. En caso de encontrar el ¿por qué? del problema se los comento. Gracias :)

**Comment by MauroMontan** [comment, marked as answer]
El WSL2 si es bastante lento, y eso sin tomar en cuenta que el CLI de vue ya es algo lento, igual tenia problemas con el wsl2, no detecta los cambios que realiza en los archivos. lo que puedes hacer es mover tu proyecto al sistema de archivos de tu ubuntu, o trabajar directamente con node en windows (en lo personal es mas comodo)

**Reply by stariluz** [reply]
Ohhh ya veo, pues si es por culpa de WSL entonces lo más sencillo si es correrlo en Windows. Mañana investigare un poco más y probare esa alternativa, muchas gracias.

**Reply by MauroMontan** [reply]
si, estoy pendiente, igual quiero saber si hay alguna solución

**Reply by stariluz** [reply]
Si la hay, ya la encontré! :D

**Reply by stariluz** [reply]
Decidí volver a checar, y veo que mi respuesta es justo lo que habías comentado 😅, en verdad una disculpa, al final me quede en mente lo de cambiarse a Windows jajaja, dejaré mi comentario por si le sirve a alguien, pero muchas gracias por la solución :)

**Comment by stariluz** [comment]
Luego de checar un rato la documentación y foros, encontré la siguiente página donde recomiendan no tener los archivos de vue bajo el sistema de archivos de windows, si no en la de el sistema de linux que usen. 
[https://docs.microsoft.com/en-us/windows/dev-environment/javascript/vue-on-wsl](url) 

Entonces volví a hacer la instalación de node, y npm (por si acaso), según la siguiente página [https://docs.microsoft.com/en-us/windows/dev-environment/javascript/nodejs-on-wsl](url) (aunque básicamente es lo mismo que ya había hecho).

Y cloné el repositorio al sistema de archivos de ubuntu, (`/home/usuario/...`), luego, dentro de el folder del proyecto,  corrí  `npm install` para que instalara las dependencias/paquetes y al correr `npm run serve` ya actualiza los cambios que realizo en los componentes.

Entonces solución: si trabajan en WSL con VUE, mantengan sus archivos afuera del /mnt/C/user, osea afuera de windows.

**Reply by stariluz** [reply]
Por cierto, eso no solo soluciona el problema, si no que también es mucho más rápido compilando. Las anteriores veces tardaba cerca de 5 minutos, y ahora solo tarda unos segundos.

**Comment by simran-sankhala** [comment]
Makes sense !
